pykechain/models/tree_traversal.py

```diff
@@ -85,14 +88,31 @@ def all_children(self: T) -> List[T]:
 
         return all_children
 
-    def count_children(self, **kwargs) -> int:
+    def count_children(self, method: Text, **kwargs) -> int:
```

Comment:
Create an .enum class for `method` and make sure the typehinting referencess that class.

Reply:
I converted this method to an abstract method to indicate a further implementation in a subclass.